workshops/face-expression-detection/README.md

```diff
@@ -0,0 +1,268 @@
+
+```
+%cd facial_expressions/
+%mkdir -p data_set/{anger,happy,neutral,sad,surprise}
+```
+
+You can see that in our cloned directory, there are some `.txt` files named as *anger.txt, happy.txt* and so on. Well these files have the names of the image files which are having those emotions/expressions in them. Meaning, an image depicting *anger* will have it's name in the *anger.txt* file. We now want to extract the images with their expressions and then save them to their respective directories which we just created. For example, the images with *anger* as their expression will go into the **anger** folder.
+
+![create-a-code-snippet](https://cloud-m37cdf42s.vercel.app/2code.png)
+
+For this, create a new code snippet and `import cv2` in it. Now we write in some basic code to move the **anger** only images from the **images** folder to **data_sets/anger**. And as a verification, we add in a `print` statement at the end.
+
+```python
+import cv2
+with open('anger.txt','r') as f:
+    img = [line.strip() for line in f]
+for image in img:
+    loadedImage = cv2.imread("images/"+image)
+    cv2.imwrite("data_set/anger/"+image,loadedImage)
+print("done writing")
+```
+
+The same will be done for all the other images. You can just change *anger.txt* from the above snippet to *happy.txt*, *sad.txt*, etc and also the `cv2` attribute to `data_set/happy/"+image,loadedImage` to this.
+
+Now, we have all the images sorted out, but we need to clear the background of these images first, so that they can then later be used for training purposes. But first, let's create a folder named **dataset** in the directory. Note that **data_sets** and **dataset** are two different directories. 
+
+```
+%mkdir dataset
+```
+
+Next, create a code snippet and `import cv2` in it. We then use some of the `cv2` attributes for renaming the all the images in a structured manner and moving the images to the **dataset** folder.
+
+```python
+import cv2
+
+with open('anger.txt','r') as f:
+    images = [line.strip() for line in f]
+
+face_detector = cv2.CascadeClassifier('haarcascade_frontalface_default.xml')
+
+# For each Emotion, enter one numeric face id
+face_id = input('\n Enter Emotion id end press <return> ==>  ')
+
+count = 0
+
+for image in images:
+    img = cv2.imread("data_set/anger/"+image)
+    gray = cv2.cvtColor(img, cv2.COLOR_BGR2GRAY)
+    faces = face_detector.detectMultiScale(gray, 1.3, 5)
+
+    for (x,y,w,h) in faces:
+
+        cv2.rectangle(img, (x,y), (x+w,y+h), (255,0,0), 2)     
+        count += 1
+
+        # Save the captured image into the datasets folder
+        cv2.imwrite("dataset/User." + str(face_id) + '.' + str(count) + ".jpg", gray[y:y+h,x:x+w])
+
+print("\n Done creating face data")
+```
```

Comment:
please try to add a line by line explanation for such complex code blocks as a beginner won't understand it easily